doc/reference/kernel/other/interrupts.rst

```diff
@@ -134,18 +134,26 @@ is running.
 
 .. important::
     The IRQ lock is thread-specific. If thread A locks out interrupts
-    then performs an operation that allows thread B to run
-    (e.g. giving a semaphore or sleeping for N milliseconds), the thread's
-    IRQ lock no longer applies once thread A is swapped out. This means
-    that interrupts can be processed while thread B is running unless
-    thread B has also locked out interrupts using its own IRQ lock.
-    (Whether interrupts can be processed while the kernel is switching
-    between two threads that are using the IRQ lock is architecture-specific.)
+    then performs an operation that puts itself to sleep (e.g. sleeping
+    for N milliseconds), the thread's IRQ lock no longer applies once
+    thread A is swapped out and the next ready thread B starts to
+    run.
+
+    This means that interrupts can be processed while thread B is
+    running unless thread B has also locked out interrupts using its own
+    IRQ lock.  (Whether interrupts can be processed while the kernel is
+    switching between two threads that are using the IRQ lock is
+    architecture-specific.)
 
     When thread A eventually becomes the current thread once again, the kernel
     re-establishes thread A's IRQ lock. This ensures thread A won't be
     interrupted until it has explicitly unlocked its IRQ lock.
 
+    If thread A does not sleep but does make a higher-priority thread B
```

Comment:
This is a good point. I'm not sure if our documentation is clear enough on this point in other relevant places; nobody should assume that the scheduling gets pended and then acted upon as soon as interrupts or the spinlock is released.

Seems like in most cases `k_yield()` is the right API to invoke the scheduler after re-enabling interrupts or releasing the lock...but I think co-op threads may not want this...Maybe for code where it's not known if it will be used in preempt or co-op context we need something like `k_yield_if_not_coop_thread()` or something like that? (which would just call z_reschedule_unlocked())

Reply:
Looks like this can be handled with:
```
	if (k_is_preempt_thread() != 0) {
		k_yield();
	}
```
without creating new API.